chia/full_node/fee_tracker.py

```diff
@@ -400,6 +411,25 @@ def estimate_median_val(
         return result
 
 
+def init_buckets() -> Tuple[List[float], SortedDict[float, int]]:
+    index = 0
+    fee_rate = 0.0
+    sorted_buckets = SortedDict()
+    buckets = []
+    while fee_rate < MAX_FEE_RATE:
+        buckets.append(fee_rate)
+        sorted_buckets[fee_rate] = index
+        if fee_rate == 0:
+            fee_rate = INITIAL_STEP
+        else:
+            fee_rate = fee_rate * STEP_SIZE
+        index += 1
+    buckets.append(INFINITE_FEE_RATE)
+    sorted_buckets[INFINITE_FEE_RATE] = index
+    assert len(sorted_buckets.keys()) == len(buckets)
+    return buckets, sorted_buckets
```

Comment:
I see how this is done now. This is where we define the bucket sizes. you don't create entries at arbitrary fee rates. This makes more sense to me now.

However, since you use a `SortedDict` (which is essentially a `std::map` or AVL-tree), it doesn't look like you need the separate `List[float]`. i.e. `buckets` is always the same as `sorted_buckets.keys()`, isn't it?

Reply:
That's correct! They don't need both. I went ahead and simplified their code - no reason to have a duplicate array.

If you look in `init_buckets`, you'll see that we do have a non-linear series making up the bucket high-water marks.